src/coreclr/vm/jitinterface.cpp

```diff
@@ -13394,8 +13413,19 @@ PCODE UnsafeJitFunction(PrepareCodeConfig* config,
         printf(".");
 #endif // _DEBUG
 
+    QueryPerformanceCounter(&qpcValue);
+    jitEndTimestamp = static_cast<int64_t>(qpcValue.QuadPart);
+
+    jitTimeQPCTicks = jitEndTimestamp - jitStartTimestamp;
+
+    FastInterlockExchangeAddLong((LONG64*)&g_cQPCTicksInJit, jitTimeQPCTicks);
+    FastInterlockExchangeAddLong((LONG64*)&g_cQPCTicksInJitForThread, jitTimeQPCTicks);
+
     FastInterlockExchangeAddLong((LONG64*)&g_cbILJitted, methodInfo.ILCodeSize);
+    FastInterlockExchangeAddLong((LONG64*)&g_cbILJittedForThread, methodInfo.ILCodeSize);
+
     FastInterlockIncrement((LONG*)&g_cMethodsJitted);
+    FastInterlockIncrement((LONG*)&g_cMethodsJittedForThread);
```

Comment:
You do not need to update the thread-local variables using interlocked operations. It slower than needed.